File: 1-patrons-de-creations/factory/code-typescript/deuxiemeMethode/fabriqueConcretes/ProduitFactory1.ts
import { ProduitFactory } from "../ProduitFactory.js";
import { Produit } from "../Produit.js";
import { Produit1 } from "../produitsConcrets/Produit1.js";

export class ProduitFactory1 extends ProduitFactory {
  constructor() {
    super();
  }

  createProduit(): Produit {
    return new Produit1();
  }
}

File: 1-patrons-de-creations/factory/code-java/deuxiemeMethode/fabrique_concretes/ProduitFactory1.java
package fabrique_concretes;

import produit_factory.ProduitFactory;
import produit.Produit;
import produits_concrets.Produit1;

public class ProduitFactory1 extends ProduitFactory {
    protected Produit createProduit() {
        return new Produit1();
    }
}